### ISSUE TITLE
error: Unsupported ONNX data type: INT64 (7)

### ISSUE THREAD

**Issue by float123** [onnx/onnx#2037]
Hello
I use torch.onnx.export() to change resnet model to onnx format and then load it in tensorRT, but error: Unsupported ONNX data type: INT64 (7);
[TensorRT] ERROR: Parameter check failed at: ../builder/Layers.cpp::ConstantLayer::1585, condition: weights.type == DataType::kFLOAT || weights.type == DataType::kHALF || weights.type == DataType::kINT32
[TensorRT] ERROR: Network must have at least one output

But I can work with ResNet50.onnx in the tensorRT samples. Why does this happen? tensorRT 5.1.2, torch 1.1.0, onnx 1.5.0

**Comment by float123** [comment]
> Hello
> I use torch.onnx.export() to change resnet model to onnx format and then load it in tensorRT, but error: Unsupported ONNX data type: INT64 (7);
> [TensorRT] ERROR: Parameter check failed at: ../builder/Layers.cpp::ConstantLayer::1585, condition: weights.type == DataType::kFLOAT || weights.type == DataType::kHALF || weights.type == DataType::kINT32
> [TensorRT] ERROR: Network must have at least one output
> 
> But I can work with ResNet50.onnx in the tensorRT samples. Why does this happen? tensorRT 5.1.2, torch 1.1.0, onnx 1.5.0

I upgraded tensorrt from 5.1.2.0 to 5.1.5.0, and the problem is solved.
WARNING: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
Successfully casted down to INT32.

**Comment by prasanthpul** [comment]
Sounds like you resolved it. Feel free to re-open if needed

**Comment by jywu-msft** [comment]
@float123 , hi I was wondering which resnet model you exported. when I tried to export pretrained resnet50 and resnet152 models from torchvision, there's a Gather node with axis=0 in the graph which TensorRT doesn't seem to support. We tried pytorch 1.1 and TensorRT 5.1.5

**Comment by float123** [comment]
> @ float123，嗨，我想知道你导出了哪个resnet模型。当我尝试从torchvision导出预训练的resnet50和resnet152模型时，TensorRT似乎不支持在图中有一个轴= 0的Gather节点。我们尝试了pytorch 1.1和TensorRT 5.1.5

I exported resnet50, may need to set the output in tensorrt:
`etwork.mark_output(network.get_layer(network.num_layers-1).get_output(0))`
But I found new problems, The ONNX Parser shipped with TensorRT 5.1.x supports ONNX IR (Intermediate Representation) version 0.0.3, opset version 9.  The onnx IR version of pytorch export is like 0.04

**Comment by wschin** [comment]
[ONNXRuntime ](https://github.com/microsoft/onnxruntime
)supports the latest ONNX operator set and maintains BC. It also has TensorRT and IntelMKL.

**Comment by tongchangD** [comment]
How do I convert INT64 to INT32
